BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -49,7 +49,7 @@ public bool IsHeartbeatEnabled
         }
 
         /// <summary>
-        /// Gets or sets the delay interval between heartbeats.
+        /// Gets or sets the delay interval between heartbeats. Setting this value will immediately reset the heartbeat timer.
```

Comment:
any way to unit test this change?

Reply:
I'm thinking about this now. I'm reviewing our existing unit test coverage. Unit tests are passing locally, but I wanted this to run on our build server to confirm that all our integration tests still pass.

Reply:
The `System.Threading.Timer` is not easily testable. This timer doesn't expose the Interval property so can't inspect that it was changed.

We don't have proper test coverage of how heartbeats are created.
I think we would need to do some significant refactoring of this class so we can manually invoke a "timer event" and validate the telemetry that gets created.
I think that's out of scope for this PR.

I propose that we take these changes as-is and revisit this later.